### PR TITLE
#4 - decode sensor data

### LINKED ISSUE #4: api endpoint for accepting sensor data

### PR DESCRIPTION
**Pull request by svnphlps**

### CODE REVIEW

File: delphi/app/routes.py [hunk 1]
```diff
@@ -1,6 +1,15 @@
...
-    return 'Hello, World!'
\ No newline at end of file
+    return 'Hello, World!'
+
+
+@app.route('/upload', methods=['POST'])
+def upload():
+    for f in request.files.getlist('file'):
+        f.save(f'uploads/{f.filename}')
+    return f'File(s) successfully uploaded'
```
Comment (svnphlps) [inline delphi/app/routes.py]: Used a f-String because it originally returned the names of the files uploaded.

File: delphi/app/routes.py [hunk 2]
```diff
@@ -1,6 +1,15 @@
 from app import app
+from flask import request
+
 
 @app.route('/')
 @app.route('/index')
 def index():
-    return 'Hello, World!'
\ No newline at end of file
+    return 'Hello, World!'
```
Comment (svnphlps) [inline delphi/app/routes.py]: Remove?

File: delphi/app/routes.py [hunk 3]
```diff
@@ -1,6 +1,23 @@
...
+
+
+@app.route('/upload', methods=['POST'])
+def upload():
+    if request.headers['content-length'] != '0' and request.headers['content-type'] in ['image/jpeg', 'image/png']:
```
Comment (svnphlps) [inline delphi/app/routes.py:15]: Only checks if the request has a Content-Type header of jpeg/png and the payload isn't empty.

File: delphi/app/routes.py [hunk 4]
```diff
@@ -1,6 +1,23 @@
...
+        buf = numpy.frombuffer(request.data, numpy.uint8)
+        current_image = cv2.imdecode(buf, cv2.IMREAD_COLOR)
```
Comment (svnphlps) [inline delphi/app/routes.py:17]: Grabs buffer from the payload and converts it into a numpy array which is then decoded into an image/matrix. Do we plan on sending images with or without color?

Comment (mooncollin) [reply delphi/app/routes.py:17]: I have only experimented with colored images. Not sure what tradeoffs we would get without color.

**Review by mooncollin — APPROVED** [review]
Looks like a good way to handle incoming data. However, we may see issues with the decoding part if the image is corrupt. This may require some exception handling near the decode part. Otherwise, looks good.

**Pull request merged**